hazardous/data/_competing_weibull.py

```diff
@@ -19,29 +21,118 @@
 )
 
 
-def _censor(y, relative_scale, random_state=None):
-    if relative_scale == 0 or relative_scale is None:
+def _censor(
+    y,
+    independent=True,
+    X=None,
+    features_censoring_rate=0.2,
+    censoring_relative_scale=1.5,
+    random_state=0,
+):
+    rng = np.random.RandomState(random_state)
+    if censoring_relative_scale == 0 or censoring_relative_scale is None:
         return y
 
-    rng = check_random_state(random_state)
-    scale = relative_scale * y["duration"].mean()
-    censoring = weibull_min.rvs(1, scale=scale, size=y.shape[0], random_state=rng)
-    y = y.copy()
-    y["event"] = np.where(y["duration"] < censoring, y["event"], 0)
-    y["duration"] = np.minimum(y["duration"], censoring)
-    return y
+    if independent:
+        scale_censoring = censoring_relative_scale * y["duration"].mean()
+        shape_censoring = 1
+    else:
+        features_impact_censoring = rng.randn(X.shape[1], 2)
+        features_impact_censoring = np.where(
+            rng.rand(
+                features_impact_censoring.shape[0], features_impact_censoring.shape[1]
+            )
+            < features_censoring_rate,
+            features_impact_censoring,
+            0,
+        )
+        df_censoring_params = censoring_relative_scale * X @ features_impact_censoring
```

Comment:
The variable names are quite hard to understand at first glance. Maybe `w_` something instead of `features_impact_censoring` and `X_params` instead of `df_censoring_params`?